BrainPortal/app/views/data_providers/show.html.erb

```diff
@@ -259,22 +264,20 @@
 
 <p>
 
-<% if (check_role(:admin_user) || @provider.user_id == current_user.id) && (@provider.is_a?(SshDataProvider) || @provider.is_a?(SmartDataProviderInterface)) %>
-
-<P>
-
-<div>
-  <table>
-    <tr><th colspan="2" class="supertitle">Public SSH Key for this CBRAIN Portal</th></tr>
-    <tr><td>This key should be installed on this Data Provider's host machine to allow remote access.</td></tr>
-    <tr><td><PRE class="ssh_key"><%= pretty_ssh_key(RemoteResource.current_resource.get_ssh_public_key || 'Unknown! Talk to sysadmin!') %></PRE></td></tr>
-  </table>
-</div>
+<% if @provider.has_owner_access?(current_user) %>
```

Comment:
Why did you remove the `if` check of the original code? It was important! Now your code will show a system SSH key even for DPs where it is irrelevant.